MinecraftClient/ChatBots/MineCube.cs

```diff
@@ -0,0 +1,340 @@
+            List<Material> liquidBlockList = new List<Material>(new Material[] { Material.Water, Material.Lava, });
+
+            var temptype = world.GetBlock(new Location(blockX, blockY + 1, blockZ)).Type;
+            var tempLoc = gravityBlockList.Contains(world.GetBlock(new Location(blockX, blockY + 1, blockZ)).Type);
+
+            return
+                // Block can not fall down on player e.g. Sand, Gravel etc.
+                gravityBlockList.Contains(world.GetBlock(new Location(blockX, blockY + 1, blockZ)).Type) ||
+                
+                // Liquid can not flow down the hole. Liquid is unable to flow diagonally.
+                liquidBlockList.Contains(world.GetBlock(new Location(blockX, blockY + 1, blockZ)).Type) ||
+                liquidBlockList.Contains(world.GetBlock(new Location(blockX - 1, blockY, blockZ)).Type) ||
+                liquidBlockList.Contains(world.GetBlock(new Location(blockX + 1, blockY, blockZ)).Type) ||
+                liquidBlockList.Contains(world.GetBlock(new Location(blockX, blockY, blockZ - 1)).Type) ||
+                liquidBlockList.Contains(world.GetBlock(new Location(blockX, blockY, blockZ + 1)).Type);
+        }
+
+        /// <summary>
+        /// Prints a whole cube to the console. Separated in layers and rows.
+        /// </summary>
+        /// <param name="cubeToPrint">Some cube</param>
+        private void PrintCubeToConsole(Cube cubeToPrint)
+        {
+            LogToConsole("Cube generated:");
+            foreach (Layer lay in cubeToPrint.LayersToMine)
+            {
+                LogToConsole("Layer:");
+                foreach (Row r in lay.RowsToMine)
+                {
+                    string generatedRow = "Row: ";
+                    foreach (Location loc in r.BlocksToMine)
+                    {
+                        generatedRow += loc.ToString() + "; ";
+                    }
+                    LogToConsole(generatedRow);
+                }
+            }
+            LogToConsole("End of cube.");
+        }
+
+        private string EvaluateMineUpCommand(string command, string[] args)
+        {
+            Location startBlock = new Location(
+                        double.Parse(args[0]),
+                        double.Parse(args[1]),
+                        double.Parse(args[2])
+                        );
+
+            Location stopBlock = new Location(
+                    double.Parse(args[3]),
+                    double.Parse(args[4]),
+                    double.Parse(args[5])
+                    );
+
+            if (Math.Round(startBlock.Y) != Math.Round(stopBlock.Y))
+            {
+                return "Both blocks must have the same Y value!";
+            }
+
+            CubeFromWorld CFW = new CubeFromWorld();
+            List<Material> materialWhitelist = new List<Material>(new Material[] { Material.Air });
```

Comment:
To populate values to a list,
```suggestion
            List<Material> materialWhitelist = new List<Material>() 
            {
                Material.Air
            };
```